src/plugins/intel_cpu/src/nodes/search_sorted.cpp

```diff
@@ -0,0 +1,116 @@
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        OPENVINO_THROW_NOT_IMPLEMENTED(errorMessage);
+    }
+    const auto ss_op = ov::as_type_ptr<const ov::op::v15::SearchSorted>(op);
+    right_mode = ss_op->get_right_mode();
+}
+
+bool SearchSorted::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (!ov::is_type<ov::op::v15::SearchSorted>(op)) {
+            errorMessage = "Only opset15 SearchSorted operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void SearchSorted::getSupportedDescriptors() {
+    // Validation is already done in the ov::opset15::SearchSorted.
+}
+
+void SearchSorted::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    ov::element::Type inputPrec = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type outputPrec = getOriginalOutputPrecisionAtPort(0);
+
+    addSupportedPrimDesc({{LayoutType::ncsp, inputPrec}, {LayoutType::ncsp, inputPrec}},
+                         {{LayoutType::ncsp, outputPrec}},
+                         impl_desc_type::ref);
+}
+
+bool SearchSorted::created() const {
+    return getType() == Type::SearchSorted;
+}
+
+bool SearchSorted::needPrepareParams() const {
+    return false;
+}
+
+void SearchSorted::executeDynamicImpl(dnnl::stream strm) {
+    execute(strm);
+}
+
+template <typename INPUT_TYPE, typename OUTPUT_TYPE>
+void SearchSorted::executeImpl() {
+    ov::reference::search_sorted<INPUT_TYPE, OUTPUT_TYPE>(getSrcDataAtPortAs<const INPUT_TYPE>(0),
+                                                          getSrcDataAtPortAs<const INPUT_TYPE>(1),
+                                                          getDstDataAtPortAs<OUTPUT_TYPE>(0),
+                                                          ov::Shape{getSrcMemoryAtPort(0)->getStaticDims()},
+                                                          ov::Shape{getSrcMemoryAtPort(1)->getStaticDims()},
+                                                          right_mode);
+}
+
+namespace {
+struct SearchSortedContext {
+    SearchSorted& node;
+};
+}  // namespace
+
+template <typename T>
+struct SearchSorted::SearchSortedExecute {
+    using TInputType = typename std::tuple_element<0, T>::type;
+    using TOutputType = typename std::tuple_element<1, T>::type;
+
+    void operator()(SearchSortedContext& ctx) {
+        ctx.node.executeImpl<TInputType, TOutputType>();
+    }
+};
+void SearchSorted::execute(dnnl::stream strm) {
+    auto inputPrecision = getParentEdgeAt(0)->getMemory().getDesc().getPrecision();
+    auto outputPrecision = getChildEdgeAt(0)->getMemory().getDesc().getPrecision();
+
+    SearchSortedContext ctx = {*this};
+
+#define CASE(OV_TYPE) OV_CASE2(OV_TYPE, ov::element::i64, ov::element_type_traits<OV_TYPE>::value_type, int64_t)
```

Comment:
OV CPU Plugin doesn't support i64 data_type for internal tensor. So current tests work only because SearchSorted op is last op in the graph: SearchSorted [i64] -> Result.
In case there is other op after SearchSorted than precision will be:  SearchSorted [i32] -> Op -> [i32] -> Convert [i64] -> Result.
So if we still need to support i64 for the first case lets handle i32 in switch below as well. Otherwise lets leave i32 only.

Reply:
i64 is needed as some models use it, so fixed by adding i32.